flang/lib/Optimizer/CodeGen/Target.cpp

```diff
@@ -126,7 +126,7 @@ struct TargetX86_64 : public GenericTarget<TargetX86_64> {
     const auto *sem = &floatToSemantics(kindMap, eleTy);
     if (sem == &llvm::APFloat::IEEEsingle()) {
       // <2 x t>   vector of 2 eleTy
-      marshal.emplace_back(fir::VectorType::get(2, eleTy), AT{});
+      marshal.emplace_back(fir::VectorType::get(eleTy.getContext(), 2, eleTy), AT{});
```

Comment:
Don't thread the context again. Should use the MLIRContext of the eleTy.

Reply:
I missed a custom builder in `FIRTypes.td` for the `VectorType`. I'll add and and reverse this change.